inugami_webapp/src/main/webapp/js/app/components/forms/input.time.slots.ts

```diff
@@ -0,0 +1,233 @@
+
+    /*****************************************************************************
+    * ATTRIBUTES
+    *****************************************************************************/
+    @Input() style                  : any;
+
+    @Input() styleClass             : string;
+
+    @Input() tabindex               : number;
+
+    @Input() inputId                : string;
+
+    @Input() name                   : string;
+
+    @Input() disabled               : boolean = false; 
+
+    @Input() readonly               : boolean = false;
+
+    @Input() labelOn                : string;
+
+    @Input() labelOff               : string;
+
+    @Input() validator              : any;
+
+    @Output() onFocus               : EventEmitter<any> = new EventEmitter();
+
+    @Output() onChange              : EventEmitter<any> = new EventEmitter();
+
+    @Output() onAdd                 : EventEmitter<any> = new EventEmitter();
+
+    @Output() onDelete              : EventEmitter<any> = new EventEmitter();
+
+    onModelChange                   : Function = () => { };
+
+    onModelTouched                  : Function = () => { };
+
+    styleClasses                    : string[];
+
+    focused                         : boolean;
+
+    valid                           : boolean = true;
+
+    enableValidator                 : boolean = false;
+
+    timeSlotsModel                  : any[] = [];
+
+    separator                       : string;
+
+    
+
+    /*****************************************************************************
+    * INIT
+    *****************************************************************************/
+
+    ngOnInit() {
+       if(this.timeSlotsModel.length == 0){
+         let initialTimeSlot = new TimeSlotBuilder().withFrom("00:00")
+                                                    .withTo("23:59")
+                                                    .build();
+         this.timeSlotsModel.push(initialTimeSlot);
+         this.separator= org.inugami.formatters.message("time.slots.to");
+       }
+    }
+    constructor() { }
+
+    /****************** ***********************************************************
+     * IMPLEMENTS ControlValueAccessor
+     *****************************************************************************/
+    writeValue(timeSlots: any): void {
+        this.timeSlotsModel = timeSlots;
+    }
+
+    registerOnChange(fn: Function): void {
+        this.onModelChange = fn;
+    }
+
+    registerOnTouched(fn: Function): void {
+        this.onModelTouched = fn;
+    }
+    setDisabledState(val: boolean): void {
+        this.disabled = val;
+    }
+
+    /*****************************************************************************
+    * METHODS
+    *****************************************************************************/
+
+    addSlot(from = "00:00", to = "23:59"){
+      if(!this.readonly && !this.disabled){
+        let newSlot = new TimeSlotBuilder().withFrom(from)
```

Comment:
Your builder can be use by another views or plugins, the best way for manipulate this one is to create an Angular Service and inject this one in your component. The root module will provide this service